docker/ubuntu.dockerfile

```diff
@@ -0,0 +1,150 @@
+# ------------------------------
+# Stage 1: Base Python Environment
+# ------------------------------
+ARG PYNE_TEST_BASE=openmc
+ARG UBUNTU_VERSION=24.04
+ARG MAKE_CORES=4
+
+FROM ubuntu:${UBUNTU_VERSION} AS base
+
+# Set environment variables
+ENV TZ=America/Chicago \
+    HOME=/root \
+    VENV_PATH=/opt/venv
+
+# Set working directory
+WORKDIR ${HOME}
+
+# Use bash shell
+SHELL ["/bin/bash", "-c"]
+
+# System setup
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+        software-properties-common \
+        python3-dev \
+        python3-venv \
+        wget \
+        build-essential \
+        git \
+        cmake \
+        gfortran \
+        libeigen3-dev \
+        libblas-dev \
+        liblapack-dev \
+        libhdf5-dev && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Create and activate virtual environment
+RUN python3 -m venv ${VENV_PATH} && \
+    ${VENV_PATH}/bin/pip install --upgrade pip && \
+    ${VENV_PATH}/bin/pip install \
```

Comment:
I never know which of these to include using `apt` and which using `pip`.  Is there a clear "rule" you have used to decide in this case?  I suppose some are not available via `apt`?